submissions/eoneoff/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env node
+
+"use strict";
+
+let help = {
+    script:`Simple port sniffer
+
+This is a simple port sniffer, which can be calles as command line script or imported as a module.
+When used from command line it takes 2 named paramentes:
+--host      obligatory parameter which defines the host address for a port scan.
+            After the --host parameter should follow the host IP addres as four numbers,
+            separated by commas or a host web address.
+
+                python sniffer.py --host www.google.com
+
+--ports     optional parementer to provide a port range for scan. The range
+            should be definded as two number with a dash between them. If not set
+            or set incorrectly the default range of (1-65535) is scanned
+
+                python sniffer.py --host wwww.google.com --ports 1024-2018
+
+--help      parameted to see this reference
+
+To import port scanner use
+
+    import sniffer.py
+
+after import you can perform scan by calling sniffer.scan()
+
+scan(host, portString, module, silent)
+
+For  detailed disambiguation of scan function use help(scan) after importing
+sniffer or call sniffer.py with --help method`
+}
+
+const dns = require('dns');

```

Comment:
Take a look at [DNS promises api](https://nodejs.org/api/dns.html#dns_dns_promises_api) to avoid promisyfying the methods in your code.

Reply:
Gosh, there is a promisified api for dns.
Too bad there is no for sockets.